### ISSUE TITLE
Negative http_req_sending values

### ISSUE THREAD

**Issue by na--** [loadimpact/k6#588]
Peter Thompson [reported in the Slack channel ](https://k6io.slack.com/archives/C3E688J8Z/p1524046923000425) that in some cases he gets negative values for `http_req_sending`:
```
    data_received..............: 57 MB  204 kB/s
    data_sent..................: 5.7 MB 20 kB/s
    group_duration.............: avg=37.44s      min=8.22s      med=32.05s  max=1m28s  p(90)=1m8s     p(95)=1m13s
    http_req_blocked...........: avg=46.02ms     min=0s         med=0s      max=2.27s  p(90)=165.91ms p(95)=245.17ms
    http_req_connecting........: avg=4.3ms       min=0s         med=0s      max=1.03s  p(90)=8ms      p(95)=9ms
    http_req_duration..........: avg=568.81ms    min=0s         med=92.06ms max=31.53s p(90)=1.16s    p(95)=2.73s
    http_req_receiving.........: avg=56.23ms     min=0s         med=0s      max=6.71s  p(90)=6.9ms    p(95)=130.94ms
    http_req_sending...........: avg=-1.184572ms min=-1.263895s med=0s      max=1.11s  p(90)=3ms      p(95)=105ms
    http_req_tls_handshaking...: avg=42.82ms     min=0s         med=0s      max=2.26s  p(90)=135.09ms p(95)=269.01ms
    http_req_waiting...........: avg=513.76ms    min=5ms        med=74.05ms max=31.53s p(90)=987.02ms p(95)=2.4s
    http_reqs..................: 2432   8.685693/s
    iteration_duration.........: avg=1m7s        min=57.55s     med=1m5s    max=1m28s  p(90)=1m19s    p(95)=1m26s
    iterations.................: 93     0.332142/s
    vus........................: 1      min=1  max=50
    vus_max....................: 50     min=50 max=50
```

Other details:
- bug occurs in both v0.20.0 and the v0.20.0-dev-dccb254 release (containing the data race fixes in `netext.Tracer`)
- https tests only
- network interruptions or timeouts may play a role

My working hypothesis is that we shouldn't save the timestamps when there's an error in all of the [`netext.Tracer`](https://github.com/loadimpact/k6/blob/master/lib/netext/tracer.go) hooks that have potential errors.

**Comment by na--** [comment]
@kiwipedro, unfortunately this bugfix didn't make it into 0.21, it would be [part](https://github.com/loadimpact/k6/blob/master/release%20notes/upcoming.md) of a new 0.21.1 bugfix release that we'll very soon push. For now, you can try building the latest `master` yourself (or using the `latest` docker image).

**Comment by kiwipedro** [comment]
Ah @na-- thanks for that.  I can wait :)

**Comment by na--** [comment]
@kiwipedro, 0.21.1 just got released: https://github.com/loadimpact/k6/releases/tag/v0.21.1

**Comment by kiwipedro** [comment]
@na-- Whoop! Thanks for the heads up

**Comment by kiwipedro** [comment]
hi @na-- still getting the problem

`PS C:\Tools\load-test> k6 version
k6 v0.21.1`

` data_received..............: 4.0 MB 83 kB/s
    data_sent..................: 113 kB 2.3 kB/s
    group_duration.............: avg=48.89s     min=48.89s     med=48.89s  max=48.89s   p(90)=48.89s   p(95)=48.89s
    http_req_blocked...........: avg=19.84ms    min=0s         med=0s      max=247.85ms p(90)=81.55ms  p(95)=187.89ms
    http_req_connecting........: avg=1.04ms     min=0s         med=0s      max=12.99ms  p(90)=5.99ms   p(95)=7.99ms
    http_req_duration..........: avg=158.17ms   min=0s         med=91.94ms max=1.34s    p(90)=333.81ms p(95)=640.23ms
    http_req_receiving.........: avg=28.86ms    min=0s         med=0s      max=924.47ms p(90)=1ms      p(95)=22.38ms
    http_req_sending...........: avg=-770.273µs min=-91.9481ms med=0s      max=184.89ms p(90)=0s       p(95)=999.2µs
    http_req_tls_handshaking...: avg=19.38ms    min=0s         med=0s      max=240.86ms p(90)=85.15ms  p(95)=191.69ms
    http_req_waiting...........: avg=130.08ms   min=5.99ms     med=91.94ms max=722.58ms p(90)=291.03ms p(95)=393.77ms
    http_reqs..................: 205    4.192472/s
    iteration_duration.........: avg=48.89s     min=48.89s     med=48.89s  max=48.89s   p(90)=48.89s   p(95)=48.89s
    iterations.................: 1      0.020451/s
    vus........................: 1      min=1 max=1
    vus_max....................: 1      min=1 max=1`

**Comment by kiwipedro** [comment]
Thanks @na-- 

Please see attached - and the results of the latest test below:



    data_received..............: 3.8 MB 64 kB/s
    data_sent..................: 96 kB  1.6 kB/s
    group_duration.............: avg=31.53s   min=31.53s       med=31.53s  max=3                                                                                                                                                                                               1.53s   p(90)=31.53s   p(95)=31.53s
    http_req_blocked...........: avg=16.56ms  min=0s           med=0s      max=2                                                                                                                                                                                               92.83ms p(90)=9.19ms   p(95)=194.28ms
    http_req_connecting........: avg=1.22ms   min=0s           med=0s      max=2                                                                                                                                                                                               0.98ms  p(90)=4.99ms   p(95)=6.99ms
    http_req_duration..........: avg=107.51ms min=5.99ms       med=36.97ms max=1                                                                                                                                                                                               .93s    p(90)=324.81ms p(95)=412.56ms
    http_req_receiving.........: avg=27.46ms  min=0s           med=0s      max=1                                                                                                                                                                                               .55s    p(90)=999.72µs p(95)=29.38ms
    http_req_sending...........: avg=641.35µs min=-65.962199ms med=0s      max=8                                                                                                                                                                                               7.94ms  p(90)=0s       p(95)=0s
    http_req_tls_handshaking...: avg=13.8ms   min=0s           med=0s      max=2                                                                                                                                                                                               62.85ms p(90)=0s       p(95)=178.09ms
    http_req_waiting...........: avg=79.4ms   min=4.99ms       med=35.97ms max=5                                                                                                                                                                                               18.7ms  p(90)=226.67ms p(95)=370.28ms
    http_reqs..................: 187    3.116623/s
    iteration_duration.........: avg=31.53s   min=31.53s       med=31.53s  max=3                                                                                                                                                                                               1.53s   p(90)=31.53s   p(95)=31.53s
    iterations.................: 1      0.016666/s
    vus........................: 1      min=1 max=1
    vus_max....................: 1      min=1 max=1`

[conFrontPage.zip](https://github.com/loadimpact/k6/files/2094720/conFrontPage.zip)

**Comment by na--** [comment]
Another k6 user [reported in slack](https://k6io.slack.com/archives/C3EDLHPTR/p1538590917000100) that `http_req_waiting` can also be negative, and I've managed to reproduce the issue with his script. I assume that the two issues are connected, but even if they're not, they should both be fixed before 1.0, so we can just use this issue for both.
